controller/setting_controller.go

```diff
@@ -804,36 +804,48 @@ func (sc *SettingController) updateLogLevel() error {
 	return nil
 }
 
-func (sc *SettingController) updateV2DataEngine() error {
-	v2DataEngineEnabled, err := sc.ds.GetSettingAsBool(types.SettingNameV2DataEngine)
+func (sc *SettingController) updateDataEngine(setting types.SettingName) error {
+	enabled, err := sc.ds.GetSettingAsBool(setting)
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "failed to get %v setting for updating data engine", setting)
 	}
 
-	if err := sc.ds.ValidateV2DataEngine(v2DataEngineEnabled); err != nil {
-		return err
+	var backednStoreDriver longhorn.BackendStoreDriverType
+	switch setting {
+	case types.SettingNameV1DataEngine:
+		backednStoreDriver = longhorn.BackendStoreDriverTypeV1
+		if err := sc.ds.ValidateV1DataEngine(enabled); err != nil {
+			return err
+		}
+	case types.SettingNameV2DataEngine:
+		backednStoreDriver = longhorn.BackendStoreDriverTypeV2
+		if err := sc.ds.ValidateV2DataEngine(enabled); err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("unknown setting %v for updating data engine", setting)
 	}
 
-	if !v2DataEngineEnabled {
-		return sc.cleanupInstanceManagerForV2DataEngine()
+	if !enabled {
+		return sc.cleanupInstanceManager(backednStoreDriver)
 	}
 
 	return nil
 }
 
-func (sc *SettingController) cleanupInstanceManagerForV2DataEngine() error {
-	imMap, err := sc.ds.ListInstanceManagersBySelectorRO("", "", longhorn.InstanceManagerTypeAllInOne, longhorn.BackendStoreDriverTypeV2)
+func (sc *SettingController) cleanupInstanceManager(backendStoreDriver longhorn.BackendStoreDriverType) error {
```

Comment:
Let's add an info log here to highlight this operation.

Reply:
Added. Thank you.